R/create_target_design.R

```diff
@@ -0,0 +1,145 @@
+#' Create target object
```

Comment:
Having this functionality work like TDM/regimen design makes a lot of sense.

however, as a user, it seems like I have to specify a lot of information in duplicate. For example, if I supply the targettype "ctrough", it seems off to then also have to supply "trough" as the offset. Having `time` default to zero also seems convenient.

what if we had one API that was "smart", i.e., recognizing that a target of "trough" or "cmax" should be offset from the next dose or the end of infusion, accordingly, and then another more flexible API for people who want to customize with a high degree of flexibility (i.e., the current form)?

Reply:
Makes sense, I think we can make the function smarter.

Reply:
It's simpler / smarter now. `time` and `offset_from` are not required, but can still be used in case they are needed for specialized circumstances.